### PR TITLE
(maint) Streamline AppVeyor settings

### PR DESCRIPTION
**Pull request by MikaelSmith**
Improve bundle install speed and robustness, git clone speed, and add
more logging.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -14,9 +16,10 @@ install:
   - ps: wget 'https://s3.amazonaws.com/kylo-pl-bucket/yaml-cpp-0.5.1-x86_64_mingw-w64_4.8.3_posix_seh.7z' -OutFile "$pwd\yaml-cpp.7z"
   - ps: 7z.exe x yaml-cpp.7z -oC:\tools | FIND /V "ing  "
 
-  - ps: (Get-Content C:\Ruby21-x64\lib\ruby\2.1.0\dl.rb) | Foreach-Object {$_ -replace "warn ", "puts "} | Set-Content C:\Ruby21-x64\lib\ruby\2.1.0\dl.rb
-  - ps: gem install bundler -q --no-ri --no-rdoc
-  - ps: bundle install --gemfile=lib/Gemfile --quiet 2>&1 | out-null
+  - gem install bundler --quiet --no-ri --no-rdoc
+  - bundle install --jobs 4 --retry 2 --gemfile=lib/Gemfile --quiet
+  - type Gemfile.lock
```
Comment (ferventcoder) [inline appveyor.yml]: Is there a Gemfile here?

Comment (ferventcoder) [inline appveyor.yml]: Might want to remove this line

File: appveyor.yml [hunk 2]
```diff
@@ -14,9 +16,10 @@ install:
...
+  - ruby -v
```
Comment (ferventcoder) [inline appveyor.yml]: Not sure what this adds if you are not using Ruby?

Comment (MikaelSmith) [inline appveyor.yml]: We are using Ruby; Facter loads it for some unit tests and the Ruby facts.

File: appveyor.yml (hunk 1 above)
Comment (MikaelSmith) [inline appveyor.yml]: It was wrong. There's a lib/Gemfile.lock, but decided I don't care about it.

File: appveyor.yml (hunk 2 above)
Comment (ferventcoder) [inline appveyor.yml]: Ah right on.

File: appveyor.yml [hunk 3]
```diff
@@ -14,9 +12,8 @@ install:
   - ps: wget 'https://s3.amazonaws.com/kylo-pl-bucket/yaml-cpp-0.5.1-x86_64_mingw-w64_4.8.3_posix_seh.7z' -OutFile "$pwd\yaml-cpp.7z"
   - ps: 7z.exe x yaml-cpp.7z -oC:\tools | FIND /V "ing  "
 
-  - ps: (Get-Content C:\Ruby21-x64\lib\ruby\2.1.0\dl.rb) | Foreach-Object {$_ -replace "warn ", "puts "} | Set-Content C:\Ruby21-x64\lib\ruby\2.1.0\dl.rb
-  - ps: gem install bundler -q --no-ri --no-rdoc
-  - ps: bundle install --gemfile=lib/Gemfile --quiet 2>&1 | out-null
+  - gem install bundler --quiet --no-ri --no-rdoc
+  - bundle install --jobs 4 --retry 2 --gemfile=lib/Gemfile --quiet
```
Comment (ferventcoder) [inline appveyor.yml]: This may require updating rubygems to do.

Comment (ferventcoder) [inline appveyor.yml]: Although the one that comes with ruby 2.1.x should be sufficient not to require it.

**Comment by MikaelSmith** [comment]
This probably doesn't really speed anything up, but removes some of the uglier kludges that weren't necessary.

**Comment by MikaelSmith** [comment]
I can incorporate other improvements from https://github.com/puppetlabs/puppet/pull/3885 once that's finalized.

File: appveyor.yml [hunk 4]
```diff
@@ -1,12 +1,10 @@
-version: 1.0.{build}
-clone_folder: c:\projects\facter
-platform:
-  - x64
+version: 3.0.0.{build}
+clone_depth: 10
```
Comment (ferventcoder) [inline appveyor.yml]: This is a tiny speed up for you :)

**Comment by peterhuene** [comment]
@MikaelSmith Is there other work we want to incorporate into this?

**Comment by MikaelSmith** [comment]
No new ideas came out of puppetlabs/puppet#3885.

File: appveyor.yml [hunk 5]
```diff
@@ -1,12 +1,10 @@
...
 install:
   - git submodule update --init --recursive
 
-  - ps: choco install mingw-w64 -y -Version 4.8.3 -source https://www.myget.org/F/puppetlabs
-  - ps: $env:PATH = "C:\Ruby21-x64\bin;C:\tools\mingw64\bin;" + $env:PATH
+  - choco install mingw-w64 -y -Version 4.8.3 -source https://www.myget.org/F/puppetlabs
+  - SET PATH=C:\Ruby21-x64\bin;C:\tools\mingw64\bin;%PATH%
 
   - ps: wget 'https://s3.amazonaws.com/kylo-pl-bucket/boost_1_55_0-x86_64_mingw-w64_4.8.3_posix_seh.7z' -OutFile "$pwd\boost.7z"
```
Comment (ferventcoder) [inline appveyor.yml]: Thanks @kylog ;)

**Comment by MikaelSmith** [comment]
I'm not aware of any holdups on merging this.

**Pull request merged**